frontend/src/rmoods/websockets/wsClient.ts

```diff
@@ -0,0 +1,46 @@
+import Cookies from "js-cookie";
+import {CreateToastFnReturn} from "@chakra-ui/react";
+import {atom} from "jotai";
+
+// Define an atom to store the WebSocket connection status
+// for now the atom manipulation is in the onerror function as it was the easiest to test on
+// later we can think about mocking the websocket connection but right now I'd focus on changing chakra version
+// because more and more components will demand bigger refactor as we develop frontend
+export const wsConnectionStatusAtom = atom<boolean>(false);
+
+class WebSocketConnection {
+  constructor(toast: CreateToastFnReturn, setWsConnectionStatus: (status: boolean) => void) {
```

Comment:
I think we've had a misunderstanding when it comes to the Jotai stuff inside WebSocketConnection.
I meant that we should be able to access any atom we have defined, not just pass a particular function for it to change.

I'll make a more detailed ticket on this matter.